operations/retyping/cast/operation.go

```diff
@@ -0,0 +1,128 @@
+package cast
+
+import (
+	"reflect"
+
+	c "github.com/spf13/cast"
+
+	"github.com/project-flogo/catalystml-flogo/action/operation"
+	"github.com/project-flogo/core/data"
+	"github.com/project-flogo/core/data/coerce"
+	"github.com/project-flogo/core/support/log"
+)
+
+type Operation struct {
+	logger log.Logger
+}
+
+var ValLen int
+var givenType data.Type
+
+func New(ctx operation.InitContext) (operation.Operation, error) {
+	return &Operation{logger: ctx.Logger()}, nil
+}
+
+func (a *Operation) Eval(inputs map[string]interface{}) (interface{}, error) {
+	in := &Input{}
+	err := in.FromMap(inputs)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Also there should be a debug logger regarding the inputs of the operation. You can review the last PR in which I added loggers.